crates/credentials/src/verifiable_credential.rs

```diff
@@ -130,6 +79,52 @@ pub struct VerifiableCredential {
     pub credential_subject: CredentialSubject,
 }
 
+impl From<VerifiableCredential> for JwtPayloadVerifiableCredential {
+    fn from(credential: VerifiableCredential) -> Self {
+        JwtPayloadVerifiableCredential {
+            context: credential.context,
+            id: Some(credential.id),
+            r#type: credential.r#type,
+            issuer: Some(credential.issuer),
+            issuance_date: Some(credential.issuance_date),
+            expiration_date: credential.expiration_date,
+            credential_subject: Some(credential.credential_subject),
+        }
+    }
+}
+
+#[derive(Serialize, Deserialize, Debug, Clone)]
+pub struct JwtPayloadVerifiableCredential {
+    #[serde(rename = "@context")]
+    context: Vec<String>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    id: Option<String>,
+    #[serde(rename = "type")]
+    r#type: Vec<String>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    issuer: Option<Issuer>,
+    #[serde(skip_serializing_if = "Option::is_none", rename = "issuanceDate")]
+    issuance_date: Option<i64>,
+    #[serde(skip_serializing_if = "Option::is_none", rename = "expirationDate")]
+    expiration_date: Option<i64>,
+    #[serde(skip_serializing_if = "Option::is_none", rename = "credentialSubject")]
+    credential_subject: Option<CredentialSubject>,
+}
+
+impl From<JwtPayloadVerifiableCredential> for VerifiableCredential {
+    fn from(payload: JwtPayloadVerifiableCredential) -> Self {
+        VerifiableCredential {
+            context: payload.context,
+            id: payload.id.unwrap_or_default(),
+            r#type: payload.r#type,
+            issuer: payload.issuer.unwrap(),
+            issuance_date: payload.issuance_date.unwrap_or_default(),
+            expiration_date: payload.expiration_date,
+            credential_subject: payload.credential_subject.unwrap(),
```

Comment:
a couple more `.unwrap()`s to kill
```suggestion
            issuer: payload.issuer.unwrap_or_default(),
            issuance_date: payload.issuance_date.unwrap_or_default(),
            expiration_date: payload.expiration_date,
            credential_subject: payload.credential_subject.unwrap_or_default(),
```

Reply:
if we don't want to use `default()` for those fields, we could return `Err` instead. In that case we would use the `TryFrom` trait instead of the `From` trait.